### PR TITLE
feat(replay): Update breadcrumbs for slow clicks and mutations

### LINKED ISSUE #50147: Add a mutation breadcrumb when limit is reached

### PR DESCRIPTION
**Pull request by billyvg**
makeover time for these two breadcrumbs to give better information

image
image
image

Closes https://github.com/getsentry/sentry/issues/50147

### CODE REVIEW

File: static/app/utils/replays/replayDataUtils.tsx [hunk 1]
```diff
@@ -179,10 +179,13 @@ export function breadcrumbFactory(
...
     .map(crumb => {
       if (crumb.category === 'replay.mutations') {
+        const crumbData = crumb.data as Record<string, unknown>;
         return {
           ...crumb,
-          type: BreadcrumbType.WARNING,
-          level: BreadcrumbLevelType.WARNING,
+          type: crumbData.limit ? BreadcrumbType.ERROR : BreadcrumbType.WARNING,
+          level: crumbData.limit
+            ? BreadcrumbLevelType.FATAL
```
Comment (bruno-garcia) [inline static/app/utils/replays/replayDataUtils.tsx:187]: I wonder if level `FATAL` is the way to go here. On mobile/desktop/server it means the process crashed usually. How about:
```suggestion
            ? BreadcrumbLevelType.ERROR
```

File: static/app/utils/replays/replayDataUtils.tsx [hunk 2]
```diff
@@ -179,10 +179,13 @@ export function breadcrumbFactory(
     })
     .map(crumb => {
       if (crumb.category === 'replay.mutations') {
+        const crumbData = crumb.data as Record<string, unknown>;
         return {
           ...crumb,
-          type: BreadcrumbType.WARNING,
-          level: BreadcrumbLevelType.WARNING,
+          type: crumbData.limit ? BreadcrumbType.ERROR : BreadcrumbType.WARNING,
```
Comment (bruno-garcia) [inline static/app/utils/replays/replayDataUtils.tsx:185]: `type` we could explore other options: https://develop.sentry.dev/sdk/event-payloads/breadcrumbs/#breadcrumb-types

Including, creating our own new type `dead-click` which has some cursor logo with a X or something

Comment (billyvg) [reply static/app/utils/replays/replayDataUtils.tsx:185]: types are really only used here to display the icon in the breadcrumb list iirc, agreed it'd be nice to have a dead click specific one!

File: static/app/utils/replays/replayDataUtils.tsx (hunk 1 above)
Comment (billyvg) [reply static/app/utils/replays/replayDataUtils.tsx:187]: tbh not sure what level even does here

File: static/app/components/replays/breadcrumbs/utils.tsx [hunk 3]
```diff
@@ -73,6 +131,14 @@ export function getTitle(crumb: Crumb) {
     return crumb.message ?? crumb.description;
   }
   const [type, action] = crumb.category.split('.') || [];
+
+  if (action === 'slowClickDetected') {
+    if ((crumb.data as Record<string, unknown> | undefined)?.endReason === 'timeout') {
```
Comment (billyvg) [inline static/app/components/replays/breadcrumbs/utils.tsx:136]: 🤮 but I want to get this out so we can more easily debug dead clicks in replays.

File: static/app/components/replays/breadcrumbs/utils.tsx [hunk 4]
```diff
@@ -73,6 +131,14 @@ export function getTitle(crumb: Crumb) {
...
+      return 'Dead Click';
+    }
+    return 'Slow Click';
```
Comment (billyvg) [inline static/app/components/replays/breadcrumbs/utils.tsx:139]: no `t()` because these are proper nouns?

**Review by ryan953 — APPROVED** [review]

**Pull request merged**